interfaces/C/Uno_C_API.cpp

```diff
@@ -96,31 +96,35 @@ class UnoModel: public Model {
    void compute_constraint_jacobian_sparsity(int* row_indices, int* column_indices, int solver_indexing,
          MatrixOrder /*matrix_order*/) const override {
       // copy the indices of the user sparsity patterns to the Uno vectors
-      std::copy_n(this->user_model.jacobian_row_indices.data(), static_cast<size_t>(this->user_model.number_jacobian_nonzeros), row_indices);
-      std::copy_n(this->user_model.jacobian_column_indices.data(), static_cast<size_t>(this->user_model.number_jacobian_nonzeros), column_indices);
+      for (size_t nonzero_index: Range(static_cast<size_t>(this->user_model.number_jacobian_nonzeros))) {
+         row_indices[nonzero_index] = this->user_model.jacobian_row_indices[nonzero_index];
+         column_indices[nonzero_index] = this->user_model.jacobian_column_indices[nonzero_index];
+      }
       // TODO matrix_order
 
       // handle the solver indexing
       if (this->user_model.base_indexing != solver_indexing) {
          const int indexing_difference = solver_indexing - this->user_model.base_indexing;
-         for (size_t index: Range(static_cast<size_t>(this->user_model.number_jacobian_nonzeros))) {
-            row_indices[index] += indexing_difference;
-            column_indices[index] += indexing_difference;
+         for (size_t nonzero_index: Range(static_cast<size_t>(this->user_model.number_jacobian_nonzeros))) {
+            row_indices[nonzero_index] += indexing_difference;
+            column_indices[nonzero_index] += indexing_difference;
          }
       }
    }
 
    void compute_hessian_sparsity(int* row_indices, int* column_indices, int solver_indexing) const override {
       // copy the indices of the user sparsity patterns to the Uno vectors
-      std::copy_n(this->user_model.hessian_row_indices.data(), static_cast<size_t>(this->user_model.number_hessian_nonzeros), row_indices);
-      std::copy_n(this->user_model.hessian_column_indices.data(), static_cast<size_t>(this->user_model.number_hessian_nonzeros), column_indices);
+      for (size_t nonzero_index: Range(static_cast<size_t>(this->user_model.number_hessian_nonzeros))) {
+         row_indices[nonzero_index] = this->user_model.hessian_row_indices[nonzero_index];
```

Comment:
@cvanaret The copy should be in the `this->user_model.hessian_row_indices` and not the contrary Charlie?!